pandas/core/dtypes/cast.py

```diff
@@ -1316,6 +1317,33 @@ def find_result_type(left_dtype: DtypeObj, right: Any) -> DtypeObj:
         #  which will make us upcast too far.
         if lib.is_float(right) and right.is_integer() and left_dtype.kind != "f":
             right = int(right)
+        # After NEP 50, numpy won't inspect Python scalars
+        # TODO: do we need to recreate numpy's inspection logic for floats too
+        # (this breaks some tests)
+        if isinstance(right, int) and not isinstance(right, np.integer):
+            # This gives an unsigned type by default
+            # (if our number is positive)
+
+            # If our left dtype is unsigned, we might not want this since
+            # this might give us 1 dtype too big, though.
+            # W should check if the corresponding int dtype (e.g. int64 for uint64)
+            # can hold the number by a little hack where
+            # we take the negative of the number
+            # If the dtype is the same size, then we will use that
+            right_dtype = np.min_scalar_type(right)
+            if right > 0 and not np.issubdtype(left_dtype, np.unsignedinteger):
+                maybe_right_dtype = np.min_scalar_type(-right)
```

Comment:
Does this properly handle floats where we may have already lost precision?

Reply:
Sorry, can you clarify?

I'm not sure how we'd get floats from here, we'd just overflow/underflow, right?

Reply:
Oh, do you mean if the left dtype is float and the right is ints? 

Lemme test.

Reply:
I tested this with floats for right (the code is only hit for integer left_dtype). It seems to work right.

Reply:
I don't think my comment applies anymore - either it refers to outdated code or I missed the isinstance check that requires this to be integral